common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch/ElasticSearchClient.scala

```diff
@@ -27,6 +30,20 @@ trait ElasticSearchClient extends ElasticSearchExecutions with GridLogging {
   def imagesCurrentAlias: String
   def imagesMigrationAlias: String
 
+  def scheduler: Scheduler
+
+  val maybeMigrationIndexName = new AtomicReference[Option[String]](None)
+
+  private val migrationIndexNameRefresher = scheduler.schedule(
+    initialDelay = 0.seconds,
+    interval = 1.minute
+  )(
+    () => {
+      // TODO what happens if this times out/fails?
```

Comment:
did you get to the bottom of this, or are we going to wait and see?

Reply:
the scheduler's scaladoc says

> If the function throws an exception the repeated scheduling is aborted, i.e. the function will not be invoked any more.

so I've updated the scheduled function to log but swallow any errors it encounters, which allows for the error to self-heal between invocations. How does this sound?